template/src/features/counter/counterSlice.js

```diff
@@ -7,22 +7,32 @@ export const slice = createSlice({
   },
   reducers: {
     increment: state => {
-      // Redux Toolkit allows us to 'mutate' the state. It doesn't actually
-      // mutate the state because it uses the immer library, which detects
-      // changes to a "draft state" and produces a brand new immutable state
-      // based off those changes
+      // Redux Toolkit allows us to write "mutating" logic in reducers. It
+      // doesn't actually mutate the state because it uses the immer library,
+      // which detects changes to a "draft state" and produces a brand new
+      // immutable state based off those changes
       state.value += 1;
     },
     decrement: state => {
       state.value -= 1;
     },
     incrementByAmount: (state, action) => {
-      state.value += action.payload.amount;
+      state.value += action.payload;
     },
   },
 });
 
+// The function below is called a selector and allows us to select a value from
+// the state. Selectors can also be defined inline where they're used instead of
+// in the slice file. For example: `useSelector((state) => state.counter.value)`
 export const selectCount = state => state.counter.value;
+
 export const { increment, decrement, incrementByAmount } = slice.actions;
 
+export const incrementAsync = () => dispatch => {
```

Comment:
Let's add a comment here saying that this is a thunk, and you can do async logic inside.

Reply:
I took a shot at adding a comment. Let me know if you want different wording